library/Booster/JsonRpc.hs

```diff
@@ -509,4 +516,11 @@ mkLogRewriteTrace
                                     , origin = Booster
                                     , result = Failure{reason = "Internal error: " <> err, _ruleId = Nothing}
                                     }
+                            ApplyEquations.SideConditionsFalse _predicates _traces ->
+                                Simplification
+                                    { originalTerm = Nothing
```

Comment:
While removing the `FIXME` that I had here, I noticed that we are actually losing traces when we hit an error after a number of successful simplifications. For `SideConditionsFalse` as well as `EquationLoop`, we would in principle have a sequence of traces that we could return to the caller for interpretation.